src/checkers/inference/InferenceAnnotatedTypeFactory.java

```diff
@@ -570,10 +571,14 @@ protected InferenceViewpointAdapter createViewpointAdapter() {
      */
     @Override
     public Set<AnnotationMirror> getTypeDeclarationBounds(TypeMirror type) {
-        final TypeElement elt = (TypeElement) getProcessingEnv().getTypeUtils().asElement(type);
```

Comment:
see if we can add a test case for this

Reply:
use breakpoint to check why we are getting non-TypeElement

Reply:
This issue is related to the change introduced in https://github.com/opprop/checker-framework/pull/207/commits/7fd273bbe10a0a229c0cce21cedeeb04fde38df0#diff-1b265befeb561d96efca87e693e6babe269d6a538df283bf1597dc0925e1f84bR339.

The original method calls `atypeFactory.getQualifierUpperBounds().getBoundQualifiers(type.getUnderlyingType())`, while the new implementation uses `atypeFactory.getTypeDeclarationBounds`. Although the latter is just a wrapper of the former in the base class, this method can be overriden by subclasses. The question is the javadoc of both methods says "Returns the set of qualifiers that are the upper bounds for a use of the type," which one should we really use?